### PR TITLE
Get all artists in a collaboration and associated recordings for better candidate sets.

### PR DESCRIPTION
**Pull request by vansika**
top_artist_df = 
| artist_credit_id       | artist_mbids    |
| :------------- | :----------: | 
|  1 | [a,b]   | 

We only have one artist.

After getting individual artists from collaboration,

top_artist_df = 

| artist_credit_id       | artist_mbids    |
| :------------- | :----------: | 
|  1 | [a,b]   | 
|  2 | [a]   | 
|  3 | [b]   | 

Three artists in top_artists_df.

More artists => more recordings => better candidate sets.

### CODE REVIEW

File: listenbrainz_spark/recommendations/candidate_sets.py [hunk 1]
```diff
@@ -124,6 +124,109 @@ def _is_empty_dataframe(df):
...
+        return res
+
+    convert_to_list_udf = func.udf(convert_to_list, ArrayType(StringType()))
+    res_df = df.withColumn("mb_artist_credit_mbids", convert_to_list_udf(col("mbids")))
+
+    return res_df
+
+
+def explode_artist_collaborations(df):
+    """ artist_mbids is a list of artist mbid of artists to which a track belongs.
```
Comment (mayhem) [inline listenbrainz_spark/recommendations/candidate_sets.py:142]: great docstring!

**Review by mayhem — APPROVED** [review]
OK, I think this all makes sense.

**Comment by vansika** [comment]
@mayhem thanks for the approval, I will push the tests.

**Comment by vansika** [comment]
@paramsingh ready to merge whenever it suits you :)

**Comment by vansika** [comment]
@paramsingh merging this now, I hope it's okay with you?

**Pull request merged**